### PR TITLE
Fix grid view button icon not showing on filesystem dock.

### LINKED ISSUE #26456: Grid View button's icon in FileSystem dock is not shown in new project
**Godot version:**
856c36c

**OS/device including version:**
Mageia 7 x86_64

**Issue description:**
When you start a new project and get the Default editor layout, the FileSystem dock is with Split Mode off. If you toggle Split Mode on, you get:
screenshot_20190301_214143

At the right of the search bar, there should be a Grid View button, but it's not visible. It is clickable though, which leads to this, as expected:
screenshot_20190301_214406

And then after toggling back to List View, the button is fixed:
screenshot_20190301_214425

**Steps to reproduce:**
See above.

**Minimal reproduction project:**
Create new project.

### PR DESCRIPTION
**Pull request by MadEqua**
Fixes #26456, which is also happening on Windows.

### CODE REVIEW

**Comment by groud** [comment]
I don't think this is the right way to fix it. 
I think the only thing to do is to set the icon when receiving NOTIFICATION_ENTER_TREE instead of the constructor.

**Comment by MadEqua** [comment]
You are right, that's a cleaner way to do it.

I've changed it, thanks for the hint.

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -302,6 +302,8 @@ void FileSystemDock::_notification(int p_what) {
 
 			always_show_folders = bool(EditorSettings::get_singleton()->get("docks/filesystem/always_show_folders"));
 
+			set_file_list_display_mode(FileSystemDock::FILE_LIST_DISPLAY_LIST);
+
 			_update_display_mode();
```
Comment (akien-mga) [inline editor/filesystem_dock.cpp]: I guess this can be removed now, as `set_file_list_display_mode` will already trigger an update.

Comment (MadEqua) [reply editor/filesystem_dock.cpp]: At first sight it seemed to me that `set_file_list_display_mode()` will update the file list display mode (thumbnails or list), while `_update_display_mode() ` will update the display mode (tree only or split).

Experimenting with it, seems that `_update_display_mode()` is actually redundant case.

Comment (akien-mga) [reply editor/filesystem_dock.cpp]: It's further down the call stack, `set_file_list_display_mode()` calls `_toggle_file_display()` which calls `_set_file_display`, which then calls `_update_file_list(true);`.

Comment (akien-mga) [reply editor/filesystem_dock.cpp]: Nevermind, I got things wrong. It doesn't actually call `_update_display_mode()`, so i guess it's safer to keep it. Sorry about that.

Comment (MadEqua) [reply editor/filesystem_dock.cpp]: Yes, it's probably more future proof to leave it there.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**